library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Types/CString.hs

```diff
@@ -0,0 +1,41 @@
+-- | Defines a type for \00 terminated strings
+module Juvix.Backends.LLVM.Codegen.Types.CString (
+  CString
+  , fromText
+  , toText
+  , length
+  , unpack
+  )
+
+where
+
+import           Data.Hashable
+import qualified Data.Text     as Text
+import           Juvix.Library hiding (length)
+import           Prelude       (String)
+import qualified Prelude
+
+newtype CString = CString Text
+  deriving (Eq, Ord, Semigroup, Monoid, Hashable)
+
+instance Show CString where
+  show = Prelude.show . toText
+
+instance ConvertText CString Text where
+  toS = toText
+
+instance ConvertText Text CString where
+  toS = fromText
+
+fromText :: Text -> CString
+fromText = CString
```

Comment:
might this be a good place to do encoding into utf8? (using a `ByteString` as the contents of a `CString` instead of a `Text`, obviously)

Reply:
I agree that `CString` is a good candidate to handle utf8 encoding. To me, it makes more sense to keep `Text` as the contents and add a function (`encodeUtf8`) that extracts and encodes the contents in utf8, returning a `ByteString`. Commit 6e66d43 implements that.

Also, the instances of `ConvertText Text CString` `ConvertText CString Text` are not inverses.
Do you think that's a problem?